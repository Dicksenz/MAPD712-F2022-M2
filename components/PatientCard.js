import React from "react";
import { View, StyleSheet, Text, TouchableOpacity } from "react-native";

import Icon from "@expo/vector-icons/MaterialCommunityIcons";
import { Colors } from "react-native/Libraries/NewAppScreen";

const PatientCard = ({ firstname, lastname, sex, age, onPress }) => {
  return (
    <TouchableOpacity onPress={onPress} style={styles.container}>
      <View style={{ display: "flex", flexDirection: "row" }}>
        <View style={styles.avatar}>
          <Icon name="account" color="#E7E7E7" size={30} />
        </View>

        <View style={styles.nameContainer}>
          <Text style={{ ...styles.nameTextStyle, marginBottom: 5 }}>
            {firstname} {lastname}
          </Text>
          <Text style={styles.subtitleStyle}>
            {sex === "M" ? "Male" : "Female"}, {age} years old
          </Text>
        </View>
      </View>

      <View style={styles.arrowCircle}>
        <Icon name="chevron-right" color="white" size={30} />
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    display: "flex",
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-between",
    backgroundColor: "white",
    paddingTop: 20,
    paddingBottom: 20,
    paddingLeft: 10,
    paddingRight: 10,
    marginBottom: 10,
    shadowColor: "#000",
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.3,
    shadowRadius: 2,
    elevation: 2,
    borderRadius: 15,
  },
  nameContainer: {
    display: "flex",
    flexDirection: "column",
  },
  nameTextStyle: {
    color: "#C4CD4A",
    fontSize: 20,
    fontWeight: "bold",
  },
  subtitleStyle: {
    color: "#A9A4A4",
    fontSize: 15,
  },

  avatar: {
    backgroundColor: "#F9F9F9",

    borderColor: "",
    padding: 10,
    borderRadius: 50,
    justifyContent: "center",
    alignItems: "center",
    marginRight: 10,
  },

  arrowCircle: {
    backgroundColor: "#347174",
    height: 30,
    width: 30,
    borderRadius: 50,
  },
});

export default PatientCard;
