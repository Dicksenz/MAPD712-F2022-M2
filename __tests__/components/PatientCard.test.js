import React from "react";
import renderer from "react-test-renderer";
import PatientCard from "../../components/PatientCard";

test("Check if PatientCard component renders correctly", () => {
  const tree = renderer.create(<PatientCard />).toJSON();
  expect(tree).toMatchSnapshot();
});
