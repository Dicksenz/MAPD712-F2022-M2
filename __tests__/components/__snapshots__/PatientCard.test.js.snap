// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if PatientCard component renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "borderRadius": 15,
      "display": "flex",
      "elevation": 2,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginBottom": 10,
      "opacity": 1,
      "paddingBottom": 20,
      "paddingLeft": 10,
      "paddingRight": 10,
      "paddingTop": 20,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.3,
      "shadowRadius": 2,
    }
  }
>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F9F9F9",
          "borderColor": "",
          "borderRadius": 50,
          "justifyContent": "center",
          "marginRight": 10,
          "padding": 10,
        }
      }
    >
      <Text />
    </View>
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#C4CD4A",
            "fontSize": 20,
            "fontWeight": "bold",
            "marginBottom": 5,
          }
        }
      >
         
      </Text>
      <Text
        style={
          Object {
            "color": "#A9A4A4",
            "fontSize": 15,
          }
        }
      >
        Female
        , 
        NaN
         years old
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#347174",
        "borderRadius": 50,
        "height": 30,
        "width": 30,
      }
    }
  >
    <Text />
  </View>
</View>
`;
